Large file uploads are now supported

acron0 opened the pull request:
* Uses the new consolidated upload commands and events
* Activity-based progression rather than event-based

mrchrisadams commented:
:sheep: it

mrchrisadams commented:
(:shipit: )

Commit:
Large file uploads are now supported

* Uses the new consolidated upload commands and events
* Activity-based progression rather than event-based

jasebell commented on src/cljs/witan/ui/ajax.cljs:
```diff
@@ -63,6 +39,18 @@
+   (merge
+    {:uri uri
+     :method :put
+     :headers {"Foo" "bar"}
```
Are we really keeping "Foo" and "bar"?

acron0 replied:
Haha, no

jasebell reviewed (approved):
One small comment, otherwise looks good.

Commit:
Adding retries into the upload loop

Commit:
Handle abandoned file uploads.

Commit:
Fixup the intercom controller

It should now cope with the new command and event styles, as well as
default values when a message is not present (as is the case with
abandoned activities)
